### ISSUE TITLE
v18.4.1 crash

### ISSUE THREAD

**Issue by Nehcgnos** [GoogleCloudPlatform/recaptcha-enterprise-mobile-sdk#90]
**Describe the bug**

SDK crash

**Integration Method**

* [ ] SPM (iOS)
* [x] Cocoapods (iOS)
* [ ] Direct Download (iOS)
* [ ] GMaven (Android)

**SDK Version (e.g. 18.0.1):**
18.4.1

**To Reproduce**

- Can not reproduce, I collected this crash via firebase

**Xcode version for iOS (please complete the following information):**
 - Xcode 15.1

**Device (please complete the following information):**
- Various devices, iOS 17, iOS 16

**Additional context**

specialized rYum3kFX.iSfZOzDK(_:)
EXC_BREAKPOINT

Crashed: riq
0  libdispatch.dylib              0x3a08c dispatch_group_leave.cold.1 + 36
1  libdispatch.dylib              0x5150 _dispatch_group_wake + 130
2  CIDER                          0x1f79e64 specialized rYum3kFX.iSfZOzDK(_:) + 473864
3  CIDER                          0x1f79eb8 specialized rYum3kFX.iSfZOzDK(_:) + 473948
4  CIDER                          0x1f75d24 closure #1 in d3q9cvwI.z5Bu9ugi(siteKey:timeout:ci5gxaVB:completionHandler:) + 457160
5  CIDER                          0x1f8130c partial apply for closure #1 in lzjQEBmQ(_:_:) + 503728
6  CIDER                          0xe064 thunk for @escaping @callee_guaranteed () -> () + 4295991396 (<compiler-generated>:4295991396)
7  libdispatch.dylib              0x26a8 _dispatch_call_block_and_release + 32
8  libdispatch.dylib              0x4300 _dispatch_client_callout + 20
9  libdispatch.dylib              0xb894 _dispatch_lane_serial_drain + 748
10 libdispatch.dylib              0xc3c4 _dispatch_lane_invoke + 380
11 libdispatch.dylib              0x17004 _dispatch_root_queue_drain_deferred_wlh + 288
12 libdispatch.dylib              0x16878 _dispatch_workloop_worker_thread + 404
13 libsystem_pthread.dylib        0x1964 _pthread_wqthread + 288
14 libsystem_pthread.dylib        0x1a04 start_wqthread + 8

**Comment by Nehcgnos** [comment]
![Uploading iShot_2024-02-21_15.15.59.png…]()

**Comment by Nehcgnos** [comment]
<img width="818" alt="iShot_2024-02-21_15 15 59" src="https://github.com/GoogleCloudPlatform/recaptcha-enterprise-mobile-sdk/assets/14212385/b4759497-011a-41ea-97c5-fed29dd36293">

**Comment by Nehcgnos** [comment]
There are also some other crashes

```
closure #1 in d3q9cvwI.czYslZw5(_:_:_:)
EXC_BREAKPOINT

Crashed: recCQ
0  libdispatch.dylib              0x3a08c dispatch_group_leave.cold.1 + 36
1  libdispatch.dylib              0x5150 _dispatch_group_wake + 130
2  CIDER                          0x1f7621c closure #1 in d3q9cvwI.czYslZw5(_:_:_:) + 458432
3  CIDER                          0x1f80b48 specialized closure #1 in rYum3kFX.init(_:_:) + 501740
4  CIDER                          0x1f80c38 partial apply for specialized closure #1 in rYum3kFX.init(_:_:) + 501980
5  CIDER                          0xe064 thunk for @escaping @callee_guaranteed () -> () + 4365115492 (<compiler-generated>:4365115492)
6  libdispatch.dylib              0x13250 _dispatch_block_async_invoke2 + 148
7  libdispatch.dylib              0x4300 _dispatch_client_callout + 20
8  libdispatch.dylib              0x77b8 _dispatch_continuation_pop + 600
9  libdispatch.dylib              0x6dd4 _dispatch_async_redirect_invoke + 584
10 libdispatch.dylib              0x15be4 _dispatch_root_queue_drain + 392
11 libdispatch.dylib              0x163ec _dispatch_worker_thread2 + 156
12 libsystem_pthread.dylib        0x1928 _pthread_wqthread + 228
13 libsystem_pthread.dylib        0x1a04 start_wqthread + 8
```

**Comment by gevorgyanvahagn** [comment]
We have the same issue after updating to 18.4.1

**Comment by walterjgsp** [comment]
Thanks for the report @Nehcgnos! We are currently working on fix for both https://github.com/GoogleCloudPlatform/recaptcha-enterprise-mobile-sdk/issues/88 and  https://github.com/GoogleCloudPlatform/recaptcha-enterprise-mobile-sdk/issues/89. I've believe that they are the same as yours.

Are you using UIKit or SwiftUI? Do you know if those crashes happen when the app is on the background or on the foreground?

**Comment by walterjgsp** [comment]
We have released version 18.4.2, which should address the issue. Please let us know if you continue to experience errors.

**Comment by Nehcgnos** [comment]
@walterjgsp Thanks reply! crash has been fixed in 18.4.2
